### PR TITLE
update a few more Python references to Python 3.10

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to https://github.com/rapidsai/build-planning/issues/88.

#4625 dropped Python 3.9 support here (as part of the RAPIDS-wide effort to drop Python 3.9 in this release). This PR fixes a few things that were missed in that PR.

### CODE REVIEW

**Review by acostadon — APPROVED** [review]
Looks good

**Review by jakirkham — COMMENTED** [review]
Thanks James! 🙏

There are some other references in the benchmarks and notebooks. That said, would leave those alone for now. Maybe we can raise an issue so the cuGraph team can follow up (as it may involve rerunning things)

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Looks like one CI job had a network failure

That said, cuDF made some changes that we need a fix for in cuGraph. This is already being handled in PR: https://github.com/rapidsai/cugraph/pull/4618

So let's wait for that to land and then pull in upstream changes to retest

**Comment by jakirkham** [comment]
Now that the aforementioned PR is in, updating the PR to include that fix and rerun CI

File: docs/cugraph/source/tutorials/cugraph_notebooks.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Running the example in these notebooks requires:
   * Download via Docker, Conda (See [__Getting Started__](https://rapids.ai/start.html))
 
 * cuGraph is dependent on the latest version of cuDF.  Please install all components of RAPIDS
-* Python 3.8+
+* Python 3.10+
 * A system with an NVIDIA GPU:  Pascal architecture or better
```
Comment (bdice) [inline docs/cugraph/source/tutorials/cugraph_notebooks.md:59]:
```suggestion
* A system with an NVIDIA GPU: Volta architecture or newer
```

File: docs/cugraph/source/tutorials/cugraph_notebooks.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Running the example in these notebooks requires:
...
 * CUDA 11.4+
 * NVIDIA driver 450.51+
```
Comment (bdice) [inline docs/cugraph/source/tutorials/cugraph_notebooks.md:61]: We've been trying to purge specific driver references from our docs.
```suggestion
```

File: notebooks/README.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ Running the example in these notebooks requires:
   * Download via Docker, Conda (See [__Getting Started__](https://rapids.ai/start.html))
 
 * cuGraph is dependent on the latest version of cuDF.  Please install all components of RAPIDS
-* Python 3.8+
+* Python 3.10+
 * A system with an NVIDIA GPU:  Pascal architecture or better
 * CUDA 11.4+
 * NVIDIA driver 450.51+
```
Comment (bdice) [inline notebooks/README.md:62]:
```suggestion
```

File: notebooks/README.md [hunk 4]
```diff
@@ -56,7 +56,7 @@ Running the example in these notebooks requires:
   * Download via Docker, Conda (See [__Getting Started__](https://rapids.ai/start.html))
 
 * cuGraph is dependent on the latest version of cuDF.  Please install all components of RAPIDS
-* Python 3.8+
+* Python 3.10+
 * A system with an NVIDIA GPU:  Pascal architecture or better
```
Comment (bdice) [inline notebooks/README.md:60]:
```suggestion
* A system with an NVIDIA GPU: Volta architecture or newer
```

**Review by bdice — APPROVED** [review]
This PR looks good but there are some other issues we could correct at the same time.

**Review by msarahan — APPROVED** [review]

**Review by alexbarghi-nv — APPROVED** [review]
👍

**Review by rlratzel — APPROVED** [review]

File: docs/cugraph/source/tutorials/cugraph_notebooks.md (hunk 2 above)
Comment (jameslamb) [reply docs/cugraph/source/tutorials/cugraph_notebooks.md]: sure! Will keep that in mind in the future.

I pushed https://github.com/rapidsai/cugraph/pull/4637/commits/02b2bbb3ab2d7c882aaaa2195e282d7c1d60e926 updating that wholegraph reference, found one other one with `git grep -i driver` too.

**Pull request merged**